docs/en/reference/testing.rst

```diff
@@ -0,0 +1,74 @@
+Testing Guidelines
+===================
+
+In order to ensure high quality, all components the Doctrine DBAL library are extensively covered with tests.
+
+Having the code covered with tests and running all tests against each individual code change helps prevent
+breakages of the library logic when its code changes.
+
+Additionally, tests:
+
+1. Help maintainers understand what problem the given code change is trying to solve.
+2. Make sure that the problem being solved needs to be solved in the DBAL.
+3. Document the proper usage of the DBAL APIs.
+
+Requirements
+------------
+
+1. Each pull request that adds new or changes the existing logic must have tests.
+2. The test that covers certain logic must fail without the changes in the code it covers.
+
+Unit Tests
+----------
+
+Unit test are meant to cover the logic of a given unit (e.g. a class or a method) including the logic of its interaction
+with other units. In this case, the other units could be mocked.
+
+Unit tests are mostly welcomed for testing the logic that the DBAL itself defines (e.g. logging). They also could be
+used as a secondary tier of testing for the logic that requires integration testing.
+
+Integration Tests
+-----------------
+
+Integration (a.k.a. functional) test are required when the behavior under the test is dictated by the logic
+defined outside of the DBAL. It could be:
+
+1. The underlying database platform.
+2. The underlying database driver.
+3. SQL syntax and the standard as such.
+
+It is important to have integration tests for the cases above. Unlike unit tests, they make the external components
+the source of truth and help make sure that the logic implemented in the DBAL is correct even if the external components
+change (e.g. a new version of a database platform is supported).
+
+When Integration Testing Is Not Required
```

Comment:
I'm not a native speaker, but this feels a bit awkward too. I don't have a better suggestion though

Reply:
I'm open to suggestions. You get the idea from the text under the header.

Reply:
"When are Integration Tests not required?"
"When Integration Tests are not required"